protocol-units/da-sequencer/node/src/whitelist.rs

```diff
@@ -0,0 +1,77 @@
+use std::{
+	collections::HashSet,
+	fs,
+	path::PathBuf,
+	sync::{Arc, RwLock},
+	thread,
+	time::Duration,
+};
+
+use ed25519_dalek::VerifyingKey;
+use hex::FromHex;
+
+#[derive(Clone)]
+pub struct Whitelist {
+	pub(crate) inner: Arc<RwLock<HashSet<VerifyingKey>>>,
+}
+
+impl Whitelist {
+	fn load(path: &PathBuf) -> std::io::Result<HashSet<VerifyingKey>> {
+		let content = fs::read_to_string(path)?;
+		let mut set = HashSet::new();
+
+		for line in content.lines() {
+			let trimmed = line.trim().trim_start_matches("0x");
+			if trimmed.is_empty() {
+				continue;
+			}
+
+			let key_bytes = <[u8; 32]>::from_hex(trimmed)
+				.map_err(|_| std::io::Error::new(std::io::ErrorKind::InvalidData, "Invalid hex"))?;
+
```

Comment:
I would to same with Error, seems that this whole file deails with std::io:: types so it's ok to import at top.